src/main/java/org/osiam/storage/query/UserFilterParser.java

```diff
@@ -50,4 +50,8 @@ public UserFilterParser(ExtensionDao extensionDao) {
         return UserQueryField.fromString(sortBy.toLowerCase(Locale.ENGLISH));
     }
 
+    @Override
+    protected UserSimpleFilterChain createFilterChain(FilterExpression<UserEntity> filter) {
+        return new UserSimpleFilterChain(entityManager.getCriteriaBuilder(), extensionDao, (UserFilterExpression) filter);
```

Comment:
Ok, this really stomps me. I don't see why have to use the cast here, but if I change the method signature to the concrete `UserFilterExpression` it doesn't work. Any hints?